pytensor/tensor/rewriting/elemwise.py

```diff
@@ -1025,3 +1028,57 @@ def local_careduce_fusion(fgraph, node):
     "fusion",
     position=49,
 )
+
+
+@node_rewriter([Elemwise])
+def inline_elemwise_scan(fgraph, node):
+    from pytensor.scan.basic import scan
+    from pytensor.scan.utils import expand_empty
+
+    scalar_op = node.op.scalar_op
+
+    if not isinstance(scalar_op, ScalarScanOp):
+        return None
+
+    # TODO: Add non-batched implementation? That should be better for scans with big difference in required n_steps
+    bcasted_inputs = broadcast_arrays(*node.inputs)
+    ret, updates = scan(
+        scalar_op.fn,
+        outputs_info=bcasted_inputs[: scalar_op.nout],
+        non_sequences=bcasted_inputs[scalar_op.nout :],
+        n_steps=scalar_op.n_steps,
+        sequences=None,
+        strict=True,
+    )
+    if updates:
+        raise ValueError("Scalar scan should never return updates")
+    if scalar_op.nout == 1:
+        ret = (ret,)
+
+    # Scan output size is given by the size of the input leading dimension, by default its n_steps + 1.
+    # If we only want to store the last elements we can shorten the leading dimension to 1
+    scan_node = ret[0].owner.inputs[0].owner
+    scan_inputs = scan_node.inputs
+    n_steps = scan_inputs[0]
+    n_non_seqs = scan_node.op.info.n_non_seqs
+    carried_inputs = scan_inputs[1 : len(scan_inputs) - n_non_seqs :]
+    constant_inputs = scan_inputs[len(scan_inputs) - n_non_seqs :]
+    new_carried_inputs = []
+    for carried_input in carried_inputs:
+        assert isinstance(carried_input.owner.op, IncSubtensor)
+        fill_value = carried_input.owner.inputs[1]
+        # TODO: Check for the global flag where this is controlled
+        new_carried_inputs.append(expand_empty(fill_value, 1))
+    ret = scan_node.op.make_node(n_steps, *new_carried_inputs, *constant_inputs).outputs
```

Comment:
One shouldn't have to hack into scan internals to generate efficient loop output without saving the intermediate results... This is needed because of #178